x-pack/legacy/plugins/ml/public/services/field_format_service.ts

```diff
@@ -89,34 +84,34 @@ class FieldFormatService {
     return fieldFormat;
   }
 
-  getFormatsForJob(jobId) {
+  getFormatsForJob(jobId: string): Promise<any[]> {
     return new Promise((resolve, reject) => {
-
       const jobObj = mlJobService.getJob(jobId);
       const detectors = jobObj.analysis_config.detectors || [];
-      const formatsByDetector = {};
+      const formatsByDetector: any[] = [];
```

Comment:
object replaced with an array 🤔

Reply:
yes, it looks like it was incorrect being an object.
`formatsByDetector` is indexed using `detector_index` from each detector which is an incrementing integer.